src/rust/scheduler/scheduler.rs

```diff
@@ -257,12 +292,38 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_scheduler_insert() -> Result<()> {
```

Comment:
We still are missing one test that ensures that if a task B that is created after the completion a task A is not assigned the same ID.